internal/internal_worker.go

```diff
@@ -1020,6 +1020,7 @@ func newAggregatedWorker(
 
 	ensureRequiredParams(&workerParams)
 	workerParams.MetricsScope = tagScope(workerParams.MetricsScope, tagDomain, domain)
+	workerParams.MetricsScope = tagScope(workerParams.MetricsScope, tagTaskList, taskList)
```

Comment:
could you update tagScope to take variable inputs so we only call tagScope once and only create one map.

Reply:
Done